crates/wast/src/wast.rs

```diff
@@ -482,3 +517,179 @@ fn v128_matches(actual: u128, expected: &wast::V128Pattern) -> bool {
         }),
     }
 }
+
+/// When troubleshooting a failure in a spec test, it is valuable to understand the bit-by-bit
+/// difference. To do this, we print a hex-encoded version of Wasm values and assertion expressions
+/// using this helper.
+fn as_hex_pattern<T>(bits: T) -> String
+where
+    T: fmt::LowerHex,
+{
+    match size_of::<T>() {
+        1 => format!("{:#04x}", bits),
+        2 => format!("{:#06x}", bits),
+        4 => format!("{:#010x}", bits),
+        8 => format!("{:#018x}", bits),
+        16 => format!("{:#034x}", bits),
+        _ => unimplemented!(),
```

Comment:
If you want to get super fancy this can all get squashed into:

```rust
format!("{1:#00$x}", size_of_val(&bits) + 2, bits)
```

[more info](https://doc.rust-lang.org/stable/std/fmt/#width)

Reply:
Sorcery... done.